pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -48,6 +48,15 @@ func (ts *TaskSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	if len(ts.Steps) == 0 {
 		errs = errs.Also(apis.ErrMissingField("steps"))
 	}
+
+	if config.IsSubstituted(ctx) {
+		// Validate the task's workspaces only.
```

Comment:
Also, I think in the linked issue, if the value passed for the parameter was '{"key": "$(workspaces.a)"}' instead of '{"key": "$(params.a)"}' (or similar) we'll run into a similar issue with workspace validation-- but maybe out of scope for this PR

Reply:
I think that the initial reason that a second validation were added was from #3786. And it ended up calling to validate all again instead of just workspaces